cmd/server-main.go

```diff
@@ -343,18 +324,23 @@ func serverMain(c *cli.Context) {
 	serverAddr := c.String("address")
 
 	// Check if requested port is available.
-	host, port := splitHostPort(serverAddr)
-	fatalIf(checkPortAvailability(port), "Port unavailable %d", port)
+	host, portStr, err := net.SplitHostPort(serverAddr)
```

Comment:
The splitHostPort job was to work when someone specifies minio server like this , perhaps it never did. 

```
minio server --address "10.1.10.1" 
``` 

Assuming port 80 in case of http and 443 in case of https. You might have to keep spitHostPort and use it to address  this issue.